### PR TITLE
Kmergenie

### PR DESCRIPTION
**Pull request by jerowe**
* [ ] I have read the guidelines above.
* [x] This PR adds a new recipe.
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

This is a recipe for kmergenie.

But I have no idea how to structure it. Help?

### CODE REVIEW

**Comment by jerowe** [comment]
OSX build is failing. It seems to be timing out.

**Comment by bgruening** [comment]
The real problem is this one on OSX:
```
+ sed -i 's/print sys.hexversion>=0x02050000/print(sys.hexversion>=0x02050000)/' makefile
sed: 1: "makefile": invalid command code m
```

OSX has a different version of sed :crying_cat_face: 
And the command needs to look like that to be compatible with both sed versions.

```
sed -i.bak 's/print sys.hexversion>=0x02050000/print(sys.hexversion>=0x02050000)/' makefile
```

**Comment by jerowe** [comment]
Why? Why does osx have to mess with sed?  Our entire infrastructure is based on find, grep and sed.

File: recipes/kmergenie/build.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+set -x -e
+
+export GCC=${PREFIX}/bin/gcc
```
Comment (bgruening) [inline recipes/kmergenie/build.sh]: This needs to go :)

**Comment by bgruening** [comment]
Now we have this one
```
make: /anaconda/envs/_build/bin/g++: No such file or directory
```
I think if you remove the exports we should be fine.

To answer you question about `sed`. Historical reasons :)
OSX is using the BSD tools and the BSD-sed whereas most of the Linux distributions are using GNU coreutils. Both are not in any case compatible. For this reason we decided in Galaxy to annotate all tools with gnu-coreutils that are using sed, awk etc... so that we can guarantee reproducible results even on OSX and BSD derivatives.

**Comment by jerowe** [comment]
```
~~Historical reasons ~~
```
```
At some point some guy through a hissy fit over something completely ridiculous
```

Fixed that for you.

**Comment by bgruening** [comment]
Let's see what supervisor Travis is saying.
Thanks for being so patient and helping us to improve our recipes!

**Comment by jerowe** [comment]
Thanks for being so patient and helping me fix my recipes. ;-)

**Comment by bgruening** [comment]
A never ending story, but we are one step closer :)

http://stackoverflow.com/questions/19242275/re-error-illegal-byte-sequence-on-mac-os-x

```
LC_ALL=C sed -i.bak 's' doobar
```

File: recipes/kmergenie/build.sh [hunk 2]
```diff
@@ -14,8 +14,9 @@ export CPPFLAGS="-I${PREFIX}/include"
 
 mkdir -p $PREFIX/bin
 
+LC_ALL=C sed -i.bak 's' doobar
```
Comment (bgruening) [inline recipes/kmergenie/build.sh]: Oh no this was just a terrible example ;)
You should probably do: `export LC_ALL=C`
And this is just a guess. Sorry for the confusing.

**Comment by bgruening** [comment]
Wuhu!

**Pull request merged**